modules/base/rendering/grids/renderablegrid.cpp

```diff
@@ -0,0 +1,320 @@
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/grids/renderablegrid.h>
+
+#include <modules/base/basemodule.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/openspaceengine.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/util/spicemanager.h>
+#include <openspace/util/updatestructures.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/glm.h>
+#include <ghoul/opengl/programobject.h>
+
+namespace {
+    constexpr const char* ProgramName = "GridProgram";
+
+    constexpr openspace::properties::Property::PropertyInfo GridColorInfo = {
+        "GridColor",
+        "Grid Color",
+        "This value determines the color of the grid lines that are rendered."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo SegmentsInfo = {
+        "Segments",
+        "Number of Segments",
+        "This value specifies the number of segments that are used to render the "
+        "grid in each direction."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineWidthInfo = {
+        "LineWidth",
+        "Line Width",
+        "This value specifies the line width of the grid."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo SizeInfo = {
+        "Size",
+        "Grid Size",
+        "This value species the size of each dimensions of the grid"
+    };
+
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderableGrid::Documentation() {
+    using namespace documentation;
+    return {
+        "RenderableGrid",
+        "base_renderable_grid",
+        {
+            {
+                GridColorInfo.identifier,
+                new DoubleVector3Verifier,
+                Optional::Yes,
+                GridColorInfo.description
+            },
+            {
+                SegmentsInfo.identifier,
+                new DoubleVector2Verifier, // TODO: Should be Int, but specification test fails...
```

Comment:
I'd suggest to write it as:
@TODO (emmbr, 2020-07-07): Should be Int, but specification test fails...

to make sure that a) who to contact in case the comment becomes less understandable over time and b) give a time estimate when we wanted this to be done properly and c) @TODO enables us to globally search for that string, the word "todo" might appear in other places